app/v2/controllers/RetrieveForeignPropertyAdjustmentsController.scala

```diff
@@ -0,0 +1,129 @@
+
+  def retrieve(nino: String, bsasId: String): Action[AnyContent] =
+    authorisedAction(nino).async { implicit request =>
+
+      val rawData = RetrieveAdjustmentsRawData(nino, bsasId)
+      val result =
+        for {
+          parsedRequest <- EitherT.fromEither[Future](requestParser.parseRequest(rawData))
+          response <- EitherT(service.retrieveForeignPropertyAdjustments(parsedRequest))
+          hateoasResponse <- EitherT.fromEither[Future](
+            hateoasFactory.wrap(response.responseData,
+              RetrieveForeignPropertyAdjustmentsHateoasData(nino, response.responseData.metadata.bsasId)).asRight[ErrorWrapper])
+        } yield {
+          logger.info(
+            s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
+              s"Success response received with correlationId: ${response.correlationId}"
+          )
+
+          auditSubmission(
+            GenericAuditDetail(
+              userDetails = request.userDetails,
+              params = Map("nino" -> nino, "bsasId" -> bsasId),
+              requestBody = None,
+              `X-CorrelationId` = response.correlationId,
+              auditResponse = AuditResponse(httpStatus = OK, response = Right(Some(Json.toJson(hateoasResponse))))
+            )
+          )
+
+          Ok(Json.toJson(hateoasResponse))
+            .withApiHeaders(response.correlationId)
+            .as(MimeTypes.JSON)
+        }
+      result.leftMap { errorWrapper =>
+        val correlationId = getCorrelationId(errorWrapper)
+        val result = errorResult(errorWrapper).withApiHeaders(correlationId)
+
+        auditSubmission(
+          GenericAuditDetail(
+            userDetails = request.userDetails,
+            params = Map("nino" -> nino, "bsasId" -> bsasId),
+            requestBody = None,
+            `X-CorrelationId` = correlationId,
+            auditResponse = AuditResponse(httpStatus = result.header.status, response = Left(errorWrapper.auditErrors))
+          )
+        )
+
+        result
+      }.merge
+    }
+
+  private def errorResult(errorWrapper: ErrorWrapper) = {
+    (errorWrapper.error: @unchecked) match {
+      case BadRequestError | NinoFormatError | BsasIdFormatError => BadRequest(Json.toJson(errorWrapper))
+      case RuleNotSelfEmployment | RuleNoAdjustmentsMade  => Forbidden(Json.toJson(errorWrapper))
+      case NotFoundError => NotFound(Json.toJson(errorWrapper))
+      case DownstreamError => InternalServerError(Json.toJson(errorWrapper))
+    }
+  }
+
+  private def auditSubmission(details: GenericAuditDetail)
+                             (implicit hc: HeaderCarrier,
+                              ec: ExecutionContext): Future[AuditResult] = {
+
+    val event = AuditEvent(
+      auditType = "retrieveBusinessSourceAccountingAdjustments",
+      transactionName = "retrieve-a-foreign-property-business-accounting-adjustments",
+      detail = details
+    )
+
+    auditService.auditEvent(event)
+  }
+}
```

Comment:
Have we confirmed that we are implementing Audits into this endpoint yet?

Reply:
good question, I'm not sure if we are

Reply:
shall I get rid of that then?

Reply:
the other one had them removed but feel free to ask james or francis about it